CMakeLists.txt

```diff
@@ -788,13 +788,14 @@ if(HPX_WITH_NETWORKING)
     OFF CATEGORY "Parcelport")
   if(HPX_WITH_PARCELPORT_MPI)
     hpx_add_config_define(HPX_HAVE_PARCELPORT_MPI)
-    if(MSVC)
-      # FIXME: add OpenMPI specific flag here for now as the hpx_add_compile_flag()
-      # below does not add the extra options to the top level directory
-      hpx_add_config_define(OMPI_IMPORTS)
-    endif()
   endif()
 
+  # we have to decide here whether we want to have MPI futures
+  hpx_option(HPX_MPI_WITH_FUTURES BOOL
+    "Enable support for returning futures from MPI asynchronous calls"
+    " (default: ON if networking+MPI already enabled)"
+    ON CATEGORY "MPI")
```

Comment:
I don't think the description matches the behaviour. Now MPI futures are on whenever networking is enabled. To keep things simple we can have the MPI parcelport and MPI futures options be independent. If one is using MPI futures one is unlikely to use the parcelport as well.

Reply:
If NETWORKING is enabled, the default for the MPI module is to be enabled as well. If NETWORKING is disabled, the default for the MPI module is to be disabled (see: https://github.com/STEllAR-GROUP/hpx/pull/4540/files/8fcea143f8d9291d2176472e06ad1b816b5d8fe2#diff-af3b638bc2a3e6c650974192a53c7291R814-R817)

Reply:
The MPI part is missing. Now `HPX_MPI_WITH_FUTURES` is `ON` if networking is on even if the MPI parcelport is off. That's why most of the pycicle configurations fail (no MPI, except for implicitly on the CUDA builds because they use the Cray compilers). I'm not sure why the configuration without networking still wants MPI though (https://cdash.cscs.ch//viewConfigure.php?buildid=104086).

Reply:
I interpret `ON if networking+MPI already enabled` as networking and MPI parcelport by the way.

Reply:
> The MPI part is missing. Now HPX_MPI_WITH_FUTURES is ON if networking is on even if the MPI parcelport is off. That's why most of the pycicle configurations fail (no MPI, except for implicitly on the CUDA builds because they use the Cray compilers). I'm not sure why the configuration without networking still wants MPI though (https://cdash.cscs.ch//viewConfigure.php?buildid=104086).

Makes sense now! I was wondering what's going on.... I'll try to fix this.